src/java.base/share/classes/jdk/internal/classfile/impl/SplitConstantPool.java

```diff
@@ -0,0 +1,604 @@
+        var e = (AbstractPoolEntry.MethodRefEntryImpl) findEntry(TAG_METHODREF, oe, ne);
+        return e == null ? internalAdd(new AbstractPoolEntry.MethodRefEntryImpl(this, size, oe, ne)) : e;
+    }
+
+    @Override
+    public InterfaceMethodRefEntry interfaceMethodRefEntry(ClassEntry owner, NameAndTypeEntry nameAndType) {
+        AbstractPoolEntry.ClassEntryImpl oe = (AbstractPoolEntry.ClassEntryImpl) owner;
+        AbstractPoolEntry.NameAndTypeEntryImpl ne = (AbstractPoolEntry.NameAndTypeEntryImpl) nameAndType;
+        if (!canWriteDirect(oe.constantPool))
+            oe = classEntry(owner.name());
+        if (!canWriteDirect(ne.constantPool))
+            ne = natEntry(nameAndType.name(), nameAndType.type());
+        var e = (AbstractPoolEntry.InterfaceMethodRefEntryImpl) findEntry(TAG_INTERFACEMETHODREF, oe, ne);
+        return e == null ? internalAdd(new AbstractPoolEntry.InterfaceMethodRefEntryImpl(this, size, oe, ne)) : e;
+    }
+
+    @Override
+    public MethodTypeEntry methodTypeEntry(MethodTypeDesc descriptor) {
+        return methodTypeEntry(utf8Entry(descriptor.descriptorString()));
+    }
+
+    @Override
+    public MethodTypeEntry methodTypeEntry(Utf8Entry descriptor) {
+        AbstractPoolEntry.Utf8EntryImpl de = maybeCloneUtf8Entry(descriptor);
+        var e = (AbstractPoolEntry.MethodTypeEntryImpl) findEntry(TAG_METHODTYPE, de);
+        return e == null ? internalAdd(new AbstractPoolEntry.MethodTypeEntryImpl(this, size, de)) : e;
+    }
+
+    @Override
+    public MethodHandleEntry methodHandleEntry(int refKind, MemberRefEntry reference) {
+        if (!canWriteDirect(reference.constantPool())) {
+            reference = switch (reference.tag()) {
+                case TAG_FIELDREF -> fieldRefEntry(reference.owner(), reference.nameAndType());
+                case TAG_METHODREF -> methodRefEntry(reference.owner(), reference.nameAndType());
+                case TAG_INTERFACEMETHODREF -> interfaceMethodRefEntry(reference.owner(), reference.nameAndType());
+                default -> throw new IllegalStateException(String.format("Bad tag %d", reference.tag()));
+            };
+        }
+
+        int hash = AbstractPoolEntry.hash2(TAG_METHODHANDLE, refKind, reference.index());
+        EntryMap<PoolEntry> map1 = map();
+        for (int token = map1.firstToken(hash); token != -1; token = map1.nextToken(hash, token)) {
+            PoolEntry e = map1.getElementByToken(token);
+            if (e.tag() == TAG_METHODHANDLE
+                && e instanceof AbstractPoolEntry.MethodHandleEntryImpl ce
+                && ce.kind() == refKind && ce.reference() == reference)
+                return ce;
+        }
+        if (!doneFullScan) {
+            fullScan();
+            return methodHandleEntry(refKind, reference);
+        }
+        return internalAdd(new AbstractPoolEntry.MethodHandleEntryImpl(this, size, hash, refKind, (AbstractPoolEntry.AbstractMemberRefEntry) reference), hash);
```

Comment:
Break the long line (same for two following methods).

Reply:
fixed, thanks.